From cc7f8fc8169ac1a30d48dc136a50cc041ba7a1ca Mon Sep 17 00:00:00 2001
From: Meticulus <theonejohnnyd@gmail.com>
Date: Sat, 3 Feb 2018 07:45:52 -0600
Subject: [PATCH] "0001-healthd-set-lcd-brightness-charger-mode
 _______________________________

*Previous From 9294dd7a53151d4ba39006ebdd209442806bd13e Mon Sep 17 00:00:00 2001
*Previous From: Meticulus <theonejohnnyd@gmail.com>
*Previous Date: Sat, 15 Apr 2017 10:42:32 -0500
*Previous Subject: [PATCH] healthd: set lcd brightness (charger mode)
*"

Change-Id: Ic3334d86e45405f41e2c2c31a47ead5c23b6bb1a
---
 healthd/healthd_mode_charger.cpp | 30 +++++++++++++++++++++++++++++-
 1 file changed, 29 insertions(+), 1 deletion(-)

diff --git a/healthd/healthd_mode_charger.cpp b/healthd/healthd_mode_charger.cpp
index 6c6d738..864050a 100644
--- a/healthd/healthd_mode_charger.cpp
+++ b/healthd/healthd_mode_charger.cpp
@@ -193,6 +193,27 @@ static int64_t curr_time_ms() {
     return tm.tv_sec * MSEC_PER_SEC + (tm.tv_nsec / NSEC_PER_MSEC);
 }
 
+void healthd_board_mode_charger_set_backlight(bool on)
+{
+    int fd;
+    char buffer[10];
+
+    memset(buffer, '\0', sizeof(buffer));
+    fd = open("/sys/class/leds/lcd_backlight0/brightness", O_RDWR);
+    if (fd < 0) {
+        LOGE("Could not open backlight node : %s\n", strerror(errno));
+        return;
+    }
+    LOGV("Enabling backlight\n");
+    snprintf(buffer, sizeof(buffer), "%d\n", on ? 102 : 0);
+    if (write(fd, buffer, strlen(buffer)) < 0) {
+        LOGE("Could not write to backlight : %s\n", strerror(errno));
+    }
+    close(fd);
+
+}
+
+
 #define MAX_KLOG_WRITE_BUF_SZ 256
 
 static void dump_last_kmsg(void) {
@@ -292,6 +313,7 @@ static void update_screen_state(charger* charger, int64_t now) {
         healthd_draw.reset(new HealthdDraw(batt_anim));
 
 #ifndef CHARGER_DISABLE_INIT_BLANK
+        healthd_board_mode_charger_set_backlight(false);
         healthd_draw->blank_screen(true);
 #endif
     }
@@ -300,6 +322,7 @@ static void update_screen_state(charger* charger, int64_t now) {
     if (batt_anim->num_cycles > 0 && batt_anim->cur_cycle == batt_anim->num_cycles) {
         reset_animation(batt_anim);
         charger->next_screen_transition = -1;
+        healthd_board_mode_charger_set_backlight(false);
         healthd_draw->blank_screen(true);
         LOGV("[%" PRId64 "] animation done\n", now);
         if (charger->charger_connected) request_suspend(true);
@@ -332,7 +355,10 @@ static void update_screen_state(charger* charger, int64_t now) {
     }
 
     /* unblank the screen  on first cycle */
-    if (batt_anim->cur_cycle == 0) healthd_draw->blank_screen(false);
+    if (batt_anim->cur_cycle == 0) {
+        healthd_board_mode_charger_set_backlight(true);
+	healthd_draw->blank_screen(false);
+    }
 
     /* draw the new frame (@ cur_frame) */
     healthd_draw->redraw_screen(charger->batt_anim, charger->surf_unknown);
@@ -430,6 +456,8 @@ static void process_key(charger* charger, int code, int64_t now) {
                    accordingly. */
                 if (property_get_bool("ro.enable_boot_charger_mode", false)) {
                     LOGW("[%" PRId64 "] booting from charger mode\n", now);
+                    healthd_board_mode_charger_set_backlight(false);
+                    gr_fb_blank(true);
                     property_set("sys.boot_from_charger_mode", "1");
                 } else {
                     if (charger->batt_anim->cur_level >= charger->boot_min_cap) {
-- 
2.7.4

